tests/integration/relations/test_backend_database.py

```diff
@@ -105,6 +107,41 @@ async def test_pgbouncer_stable_when_deleting_postgres(ops_test: OpsTest):
             ),
         )
 
-        await ops_test.model.applications[PG].remove()
-        wait_for_relation_removed_between(ops_test, PG, PGB)
-        await ops_test.model.wait_for_idle(apps=[PGB], status="active", timeout=1000)
+        await scale_application(ops_test, PGB, 3)
+        await asyncio.gather(
+            ops_test.model.wait_for_idle(
+                apps=[PGB], status="active", timeout=1000, wait_for_exact_units=3
+            ),
+            ops_test.model.wait_for_idle(
+                apps=[PG], status="active", timeout=1000, wait_for_exact_units=3
```

Comment:
Just out of curiosity, do you know if `wait_for_exact_units` works if we provide multiple apps in the same `wait_for_idle` call?

Reply:
I think it does, but I believe it waits for the _total_ units - in this case, six. I think this is cleaner, since we know we want 3 units of PGB, and 3 of PG.

Reply:
Thanks for the information!